rust/openvasctl/src/pref_handler.rs

```diff
@@ -0,0 +1,626 @@
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use models::{AliveTestMethods, Credential, Port, PortRange, Scan};
+
+    use super::PreferenceHandler;
+    use crate::openvas_redis::{FakeRedis, KbAccess};
+
+    #[test]
+    fn test_prefs() {
+        let mut scan = Scan::default();
+        scan.scan_id = Some("123-456".to_string());
+        scan.target.alive_test_methods = vec![AliveTestMethods::Icmp, AliveTestMethods::TcpSyn];
+        scan.target.credentials = vec![Credential {
+            service: models::Service::SSH,
+            port: Some(22),
+            credential_type: models::CredentialType::UP {
+                username: "user".to_string(),
+                password: "pass".to_string(),
+            },
+        }];
+        scan.target.excluded_hosts = vec!["127.0.0.1".to_string()];
+        scan.target.ports = vec![Port {
+            protocol: Some(models::Protocol::TCP),
+            range: vec![
+                PortRange {
+                    start: 22,
+                    end: Some(25),
+                },
+                PortRange {
+                    start: 80,
+                    end: None,
+                },
+            ],
+        }];
+
+        let rc = FakeRedis {
+            data: HashMap::new(),
+        };
+
+        let mut prefh = PreferenceHandler::new(scan, rc);
+        assert_eq!(prefh.redis_connector.kb_id().unwrap(), 3);
+        assert!(prefh.prepare_scan_id_for_openvas().is_ok());
+        assert!(prefh
+            .redis_connector
+            .item_exists("internal/scanid", "123-456"));
+        assert!(prefh.redis_connector.item_exists("internal/123-456", "new"));
+
+        assert!(prefh.prepare_main_kbindex_for_openvas().is_ok());
+        assert!(prefh
+            .redis_connector
+            .item_exists("internal/123-456/scanprefs", "ov_maindbid|||3"));
+
+        assert!(prefh.prepare_boreas_alive_test().is_ok());
+        assert!(prefh
+            .redis_connector
+            .item_exists("internal/123-456/scanprefs", "ALIVE_TEST|||3"));
+
+        assert!(prefh.prepare_host_options_for_openvas().is_ok());
+        assert!(prefh
+            .redis_connector
+            .item_exists("internal/123-456/scanprefs", "excluded_hosts|||127.0.0.1"));
+
+        assert!(prefh.prepare_credentials_for_openvas().is_ok());
+        assert!(prefh.redis_connector.item_exists(
+            "internal/123-456/scanprefs",
+            "1.3.6.1.4.1.25623.1.0.103591:3:password:SSH password (unsafe!):|||pass"
+        ));
+        assert!(prefh.redis_connector.item_exists(
+            "internal/123-456/scanprefs",
+            "1.3.6.1.4.1.25623.1.0.103591:1:entry:SSH login name:|||user"
+        ));
+
+        assert!(prefh.prepare_ports_for_openvas().is_ok());
+        assert!(prefh
+            .redis_connector
+            .item_exists("internal/123-456/scanprefs", "PORTS|||tcp:22-25,80,"));
+
+        println!("{:?}", prefh.redis_connector.data);
```

Comment:
This println seems odd. Was it meant to be rewritten as an assert?

Reply:
removed with [6e68d1f](https://github.com/greenbone/openvas-scanner/pull/1567/commits/6e68d1fffc1a18665d4041055ffcf45ce3fa49c9)